pkg/vgmanager/vgmanager_controller.go

```diff
@@ -394,12 +397,85 @@ func (r *VGReconciler) addDevicesToVG(vgName string, devices []internal.BlockDev
 }
 
 // filterMatchingDevices returns unmatched and matched blockdevices
-// TODO: Implement this
-func filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
-	// currently just match all devices
+func (r *VGReconciler) filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
+
+	var filteredBlockDevices []internal.BlockDevice
+
+	if volumeGroup.Spec.DeviceSelector != nil && len(volumeGroup.Spec.DeviceSelector.Paths) > 0 {
+		vgs, err := ListVolumeGroups(r.executor)
+		if err != nil {
+			return []internal.BlockDevice{}, []internal.BlockDevice{}, fmt.Errorf("failed to list volume groups. %v", err)
+		}
+
+		for _, path := range volumeGroup.Spec.DeviceSelector.Paths {
+
+			diskName, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				err = fmt.Errorf("unable to find symlink for disk path %s: %v", path, err)
+				return []internal.BlockDevice{}, []internal.BlockDevice{}, err
+			}
+
+			isAlreadyExist := isDeviceAlreadyAddedtoVG(vgs, diskName, volumeGroup)
+			if isAlreadyExist {
+				continue
+			}
+
+			blockDevice, ok := hasExactDisk(blockDevices, diskName)
+
+			if filepath.Dir(path) == internal.DiskByPathPrefix {
+				// handle disk by path here such as /dev/disk/by-path/pci-0000:87:00.0-nvme-1
+				if ok {
+					blockDevice.DiskByPath = path
+					filteredBlockDevices = append(filteredBlockDevices, blockDevice)
+				} else {
+					err = fmt.Errorf("can not find disk path %s disk name %s in the available block devices", path, diskName)
+					return []internal.BlockDevice{}, []internal.BlockDevice{}, err
+				}
+			} else if filepath.Dir(path) == internal.DiskByNamePrefix {
+				// handle disk by names here such as /dev/nvme0n1
+				if ok {
+					filteredBlockDevices = append(filteredBlockDevices, blockDevice)
+				} else {
+					err := fmt.Errorf("can not find disk name %s in the available block devices", path)
```

Comment:
```suggestion
					err := fmt.Errorf("can not find disk name %s in the available block devices", diskName)
```

Reply:
both will be the same only, So updating to path should make any differences